### ISSUE TITLE
OSHI will change to MIT license in 2019

### ISSUE THREAD

**Issue by dbwiddis** [oshi/oshi#593]
**TL;DR**

OSHI will be relicensed under the MIT License on or after January 1, 2019, unless significant objections are raised.  If you have contributed to the project:
- If you concur, thumbs-up this post (preferred rather than a "+1" comment) and/or check the box by your login, or feel free to offer positive commentary below.
- If you disagree with this course of action, please let us know in the comments below.

**What is changing?**

OSHI is changing its license from [EPL version 1.0](https://opensource.org/licenses/EPL-1.0) to the [MIT License](https://opensource.org/licenses/mit-license.php) concurrent with the first project commit in 2019.

**Why is this changing?**

Broadly speaking, there are two categories of open source/free licenses: Permissive, and Copyleft.  Permissive licenses allow recipients much more freedom with the code, including relicensing, while copyleft licenses require preservation of the original license either for the portion of the code copied (weak copyleft) or the entire project (strong copyleft).  The EPL is a weak copyleft license, which makes it incompatible with strong copyleft licenses such as the GPL.  

In specific relevance to this project, it would not be legally permissible for GPL licensed software to include OSHI, even as a library.  Changing OSHI's license from a copyleft (EPL) to a permissive (MIT) license will increase the number of projects which can use it.

Our motivation is in line with industry trends towards permissive licenses, where developers want their work to be useful.
![image](https://user-images.githubusercontent.com/9291703/44704261-909ac300-aa4f-11e8-9812-55c93d477f57.png)

**Do all committers need to agree to this change?**

Probably. Or at least have the opportunity to object, which this Issue is. I haven't consulted a lawyer and don't plan to, and am trusting that most contributors will agree.  In the interest of the open source community, I'm asking even if not required, and hoping to get affirmative answers from all contributors.  I'll consider non-response or objections on a case-by-case basis.

Also, this is a cool opportunity for me to thank everyone that's contributed to the project over the years!

**What if I don't agree for you to relicense my contribution to the project?**

I'll either keep the project license as-is, or remove your contribution before relicensing.  But I'd love to hear your reasoning to help me decide which of those two options is better.

**What if I don't care what you do with my contribution?**

Please at least thumbs-up this comment so I don't spend time fretting over whether you'll chase me down sometime in 2019 with a high powered lawyer suing me for violating your rights. ;)

**Why January 2019?**

Because the incredibly hard work involved in changing the license header in every single file, which will take me at least 2 or 3 minutes thanks to Maven, will probably occur anyway with the change of year.  And that timeframe leaves ample time for committers to comment.

**Why the MIT License?**

[It's the most common license](https://resources.whitesourcesoftware.com/blog-whitesource/open-source-licensing-trends-2017-vs-2016), is simple and to the point, and I don't have a good reason to choose anything else.  

**Did you consider other licenses?**

Yes, particularly Apache 2.0, the second most common license, and also permissive; Apache/LGPL dual to match JNA, which we leverage extensively and to which we've contributed a lot of OSHI's code; or EPL2.0 dual with GPL which is the next version of our current license that would resolve the compatibility issue.  When it got to the point that I wanted to talk to a lawyer, I stopped considering and stuck with the original author's preference for MIT.  (I did consider [WTFPL](http://www.wtfpl.net/) but figured MIT was pretty close.)

**Who are the awesome people who have contributed to this project?**

You can see them [here](https://github.com/oshi/oshi/graphs/contributors) but since I want to tag them so that they see this post, here's the complete list (with checkmarks to track concurrence for the change):

- [x] @dbwiddis
- [x] @dblock
- [x] @hazendaz
- [x] @henryx
- [x] @chikei
- [x] @le-yams
- [ ] @lundefugl
- [x] @laurent-r
- [x] @alessiofachechi
- [x] @cjbrowne
- [ ] @michaeldesigaud
- [x] @kamenitxan
- [ ] @plamenko
- [ ] @khmarbaise
- [x] @spyhunter99
- [x] @bildechinger
- [x] @jpbempel
- [x] @jsimomaa
- [x] @darinhoward
- [ ] @lprimak
- [ ] @ckadner
- [ ] @NagyGa1
- [x] @cweitkamp
- [x] @swimmesberger
- [ ] @detyo
- [x] @Sorceror
- [x] @YoshiEnVerde
- [ ] @dustin-johnson
- [x] @lu-ko
- [ ] @MarcMil
- [x] @cilki
- [x] @Kald0
- [x] @UnAfraid
- [ ] @brettwooldridge
- [x] @dpagano
- [x] @nsydenham
- [ ] @muryoh
- [x] @icirellik
- [x] @Krillsson
- [ ] @flake9025
- [ ] @stefan-reich
- [ ] @tpalusga

_(Note: the mouseover won't display more than 10 thumbsups.  [See here](https://stackoverflow.com/questions/36641368/get-reactions-using-the-github-api) for how to list via the api.)_

**Comment by lu-ko** [comment]
You do a great job. Go on!

**Comment by dblock** [comment]
@dbwiddis My general comment is that you've done an _amazing_ job carrying this project forward and am so happy to see it relicensed under MIT. This should save a lot of people wasted energy having to clear yet another strange license (EPL) with yet another lawyer in their dark corporate cubicle worlds.

**Comment by dbwiddis** [comment]
A reminder to those who haven't responded yet:

OSHI will be relicensed under the MIT License on or after January 1, 2019, unless significant objections are raised. If you have contributed to the project:

- If you concur, thumbs-up the first entry on this post (preferred rather than a "+1" comment) and/or check the box by your login, or feel free to offer positive commentary below.
- If you disagree with this course of action, please let us know in the comments below.

Still hoping for thumbs up from: @lundefugl , @laurent-r , @michaeldesigaud , @plamenko , @khmarbaise , @jpbempel , @jsimomaa , @lprimak , @ckadner , @NagyGa1 , @detyo , @dustin-johnson , @MarcMil , @brettwooldridge , @nsydenham , @muryoh , @flake9025 , @stefan-reich , @tpalusga
